### PR TITLE
Add Django 1.11 compatibility, and other modernization

### PR DESCRIPTION
**Pull request by schbetsy**
- [x] Add Django 1.11 compatibility
- [x] Upgrade requests to 2.18.X, to match [cfgov-refresh](https://github.com/cfpb/cfgov-refresh/blob/master/requirements/libraries.txt#L33)
- [x] Use Django migrations to replace south for handling db migrations
- [x] Use Tox to test across Django versions 1.8 and 1.11
- [x] Put requirements in `setup.py` under `install_requires` and `testing_extras`
- [x] Rename the default testing database from `oah` to `oah.sqlite3`, for clarity

This required upgrading `djangorestframework` to `3.6.3` or higher.

### CODE REVIEW

**Comment by schbetsy** [comment]
@chosak I was getting errors from the migration commands until I ran `pip install psycopg2-binary` in my environment. Is that a concern?

File: setup.py [hunk 1]
```diff
@@ -12,6 +12,24 @@ def read_file(filename):
         return ''
 
 
+install_requires = [
+    'beautifulsoup4==4.5.3',
```
Comment (schbetsy) [inline setup.py]: Should these requirements stay pinned to specific versions, like they were in the `requirements.txt`, or do we want them to be flexible?

File: setup.py [hunk 2]
```diff
@@ -12,6 +12,24 @@ def read_file(filename):
...
+    'Django>=1.8,<1.12',
+    'django-cors-headers',
+    'dj-database-url==0.4.2',
+    'django-localflavor',
+    'djangorestframework==3.1.3',
+    'requests>2.18,<2.20',
```
Comment (schbetsy) [inline setup.py]: All tests passed when I tested over 2.18 and 2.19.

**Comment by schbetsy** [comment]
@chosak Thanks for your help and patience. This is ready for review again.

**Comment by schbetsy** [comment]
Correct. This app works fine on drf 3.1.3 when it uses django 1.8 (that's what it was on before this PR). For django 1.11, it requires drf 3.6.3 or higher. 

I'm not sure what you're suggesting. Should I not be specifying a version of drf in setup.py since it will be ignored anyway? Should I make it a version range (e.g. `djangorestframework>=3.1.3`) and let it pull in the version it thinks it needs?

**Comment by chosak** [comment]
Sorry for being unclear: this PR is great as-is.

To me it seems like it _should_ cause a conflict for cfgov-refresh and satellite apps to have different dependencies, but as it happens, it doesn't, and cfgov-refresh overrides based on how the build/deploy process happens. That means that right now, cfgov-refresh's 3.1.3 will be used in builds, even though, after this PR, oah-api will be requiring 3.6.3.

I was suggesting that we could bump the version of oah-api in cfgov-refresh and leave DRF alone there (although I do think as part of that bump we should add `--fake-initial` to the refresh script). We can tackle a cfgov-refresh DRF upgrade separately.

**Pull request merged**